### PR TITLE
Start frontend

### PR DESCRIPTION
**Pull request by samyuh**
How to run: [alternatives]
- `docker-compose` in root
-  `run_docker.sh` in root

Hosts:
- React Server: localhost:3000
- Solr Admin: localhost:8983

### CODE REVIEW

**Review by Jumaruba — CHANGES_REQUESTED** [review]
- It's not recommended to use `COPY . .` in `dockerfile`. On this way, all the project must be build again when a small modification in `package.json` is done. It should be:
```docker
COPY package*.json ./
COPY src/ src/
COPY public/ public/ 
```
If the package*.json is the only thing that has changed in the project, the `docker` will just update it instead of the whole project.  
Also, by doing `COPY . .` the `node_modules` is being added to the `docker`. 

- Not possible to execute the program.   
There're some compilation problems: 

```
Compiled with problems:

ERROR in ./node_modules/@mui/styled-engine/GlobalStyles/GlobalStyles.js 3:0-40

Module not found: Error: Can't resolve '@emotion/react' in '/node_modules/@mui/styled-engine/GlobalStyles'


ERROR in ./node_modules/@mui/styled-engine/StyledEngineProvider/StyledEngineProvider.js 3:0-47

Module not found: Error: Can't resolve '@emotion/react' in '/node_modules/@mui/styled-engine/StyledEngineProvider'


ERROR in ./node_modules/@mui/styled-engine/index.js 6:0-39

Module not found: Error: Can't resolve '@emotion/styled' in '/node_modules/@mui/styled-engine'


ERROR in ./node_modules/@mui/styled-engine/index.js 30:0-62

Module not found: Error: Can't resolve '@emotion/react' in '/node_modules/@mui/styled-engine'


ERROR in ./node_modules/@mui/system/esm/ThemeProvider/ThemeProvider.js 11:27-60

export 'ThemeContext' (imported as 'StyledEngineThemeContext') was not found in '@mui/styled-engine' (possible exports: GlobalStyles, StyledEngineProvider, default)


ERROR in ./node_modules/@mui/system/esm/index.js 1:0-88

export 'css' (reexported as 'css') was not found in '@mui/styled-engine' (possible exports: GlobalStyles, StyledEngineProvider, default)


ERROR in ./node_modules/@mui/system/esm/index.js 1:0-88

export 'keyframes' (reexported as 'keyframes') was not found in '@mui/styled-engine' (possible exports: GlobalStyles, StyledEngineProvider, default)
```

File: frontend/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM node:latest
+
+WORKDIR .
+COPY package.json ./
+RUN npm install
+
+# Copy project content to docker  
+COPY package*.json ./
+COPY src/ src/ 
+COPY public/ public/
```
Comment (Jumaruba) [inline frontend/Dockerfile]: Added individual `COPY` to avoid adding `node_module` and improve build performance.

File: docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+version: "3"
+
+services:
+  frontend:
+    build: ./frontend
+    ports:
+      - 3000:3000
+    depends_on:
+      - solr
```
Comment (Jumaruba) [inline docker-compose.yaml:9]: Missing volumes to allow live change in the code.

**Review by Jumaruba — APPROVED** [review]
The last commits updates the requested changes.

**Comment by dianaamfr** [comment]
Deprecated warnings:
image

Fetch error due to CORS policy (so I couldn't test the search system in the frontend):
image

I think the problem is with Solr.

**Review by dianaamfr — APPROVED** [review]

**Pull request merged**